code/model/VirtualPage.php

```diff
@@ -510,8 +497,17 @@ public function getViewer($action) {
 	 * We can't load the content without an ID or record to copy it from.
 	 */
 	public function init(){
-		if(isset($this->record) && $this->record->ID){
-			if($this->record->VersionID != $this->failover->CopyContentFrom()->Version){
+		if (isset($this->record)) {
+			if (is_array($this->record)) {
```

Comment:
I expect that this is a bug; it would be good to fix the bug but outside the scope of the PHP7 work.

I recommend:

 *  calling a PHP5.6-based or `3` mainline test run and seeing if the inner block gets called. I want to confirm that `ViewableData::__get() / __isset()` changes haven't mucked with this.

 * commenting out the block and seeing if tests pass. If they do, I'd raise a separate ticket for a bugfix that replaces `&& $this->record->ID` with `&& $this->record['ID']`.`$this->record` should never be an object and we don't need to test for that. I suggest making it a separate PR/issue since we'd probably want to include a test for it and get it merged to SS4, both of which are a distraction from the task at hand.

That being the case, I think we can move forward with the code commented out with a link to the aforementioned ticket.